Gems/Smoothing/CMakeLists.txt

```diff
@@ -0,0 +1,6 @@
+
+o3de_gem_setup("Smoothing")
```

Comment:
I tested this PR on `point-release/23103` branch and I've had some problems with this line - I've had to copy something like this from some other gem (I believe that this is copied from RobotecSpectatorCamera gem):


```
o3de_find_ancestor_gem_root(gem_path gem_name "${CMAKE_CURRENT_SOURCE_DIR}")
if (NOT gem_name)
    set(gem_name "Smoothing")
endif()

if (NOT gem_path)
    set(gem_path ${CMAKE_CURRENT_SOURCE_DIR})
endif()

set(gem_json ${gem_path}/gem.json)

o3de_restricted_path(${gem_json} gem_restricted_path gem_parent_relative_path)

o3de_pal_dir(pal_dir ${CMAKE_CURRENT_SOURCE_DIR}/Platform/${PAL_PLATFORM_NAME} "${gem_restricted_path}" "${gem_path}" "${gem_parent_relative_path}")
```


I am not sure if above code is forward compatible, but it appears that `o3de_gem_setup` is not backward compatible or branch, that I've used to test this PR, is defective.